### PR TITLE
Music extension

### PR DESCRIPTION
**Pull request by rawnly**
## Description
Little extension (like the spotify one) to handle media controls for the Music app.

## Type of change
- New extension

## Screencast
CleanShot 2021-11-13 at 17 57 25

## Checklist
- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I checked that extension's icon has size of `512x512`

### CODE REVIEW

File: extensions/music/package.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "music",
+  "title": "Music",
+  "description": "Manage apple music",
```
Comment (fe9lix) [inline extensions/music/package.json:5]:
```suggestion
  "description": "Control the Apple Music app with your keyboard.",
```

File: extensions/music/package.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    {
+      "name": "toggle-play-pause",
+      "title": "Toggle Play/Pause",
+      "description": "Toggle play or pause of current track in Music.",
+      "mode": "no-view"
+    },
+    {
+      "name": "play",
+      "title": "Play",
+      "description": "Plays current track in Music.",
```
Comment (fe9lix) [inline extensions/music/package.json:24]:
```suggestion
      "description": "Play current track in Music.",
```

File: extensions/music/package.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    {
+      "name": "pause",
+      "title": "Pause",
+      "description": "Pause current track in Music.",
+      "mode": "no-view"
+    },
+    {
+      "name": "next-track",
+      "title": "Next Track",
+      "description": "Skips to the next track in Music.",
```
Comment (fe9lix) [inline extensions/music/package.json:36]:
```suggestion
      "description": "Skip to the next track in Music.",
```

File: extensions/music/package.json [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      "mode": "no-view"
+    },
+    {
+      "name": "previous-track",
+      "title": "Previous Track",
+      "description": "Skips to the previous track in Music.",
```
Comment (fe9lix) [inline extensions/music/package.json:42]:
```suggestion
      "description": "Skip to the previous track in Music.",
```

File: extensions/music/src/util/controls.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+export interface TrackInfo {
+	name: string;
+	artist: string;
+	album: string;
+	duration: string;
+	state: PlayerState
+}
+
+export const getCurrentTrack = (): TE.TaskEither<Error, Readonly<TrackInfo>> => {
```
Comment (fe9lix) [inline extensions/music/src/util/controls.ts:37]: Is that function used somewhere?

**Review by fe9lix — CHANGES_REQUESTED** [review]
Hey @Rawnly – thanks for the contribution. I've left some smaller comments...

**Comment by fe9lix** [comment]
Hey @Rawnly – is the PR finished from your side?
I was wondering about the code around getting the "Current Track" info, similar to the Spotify commands.
I guess this should either be a new command or the code should be removed.

**Comment by Rawnly** [comment]
@fe9lix I'm going to remove it in a few minutes! Sorry for delay 😅

**Comment by Rawnly** [comment]
I'm going to push the latest in a few minutes! I've added a new command for it!

**Review by fe9lix — APPROVED** [review]

**Pull request merged**